### PR TITLE
Shorthand Syntax, plus 'between' and 'distinct' filtering

### PR DESCRIPTION
**Pull request by pmccarren**
This PR covers a couple issues:
- Introducing the concept of Shorthand (esbenp/bruno#23)
- Adds the ability to execute 'distinct' queries (esbenp/bruno#22)
- Adds the ability to filter using 'between' method (esbenp/bruno#21)

I welcome any comments + feedback!
Thanks!

### CODE REVIEW

**Comment by w0rd-driven** [comment]
There was a pull request (#10) that added the between operator to the 2.1.0 tag but I don't seem to see the code in master/3.0.0. I don't know the best place to point out something like this, so sorry if it seems out of place. I would make an issue but it would involve saying something like "accept this PR" since I feel everything here is worth including.

**Comment by pmccarren** [comment]
@w0rd-driven good catch, and definitely not out of place!

Looks like `between` support was half implemented in 846537e. This implementation would be have been broken because we need to specify a method change rather than change the operator.
It looks like `between` support was then removed in 1816755.

6a555bc brings it back and handles the method change accordingly (identical to how `in` is handled)

**Comment by Carghaez** [comment]
Why implement "between" operator when you can have the same functionality through the combination of:
- gte + lte
- gt + lte
- gte + lt
- gt + lt

Anyway commits seems to be ok :D looks good!

**Comment by pmccarren** [comment]
Haha you are absolutely correct! At the end of the day I would say the 'between' operator is more a preference, though I'm partial to the added flexibility

Thanks for taking a look at it!

**Comment by esbenp** [comment]
Thank you for your contribution @pmccarren ! Sorry I have not gotten around to look at it yet, but will do very soon!

**Comment by pmccarren** [comment]
No problem at all @esbenp!

File: README.md [hunk 1]
```diff
@@ -305,6 +308,51 @@ Books with authors whoose name start with `Optimus` or ends with `Prime`.
...
+```
+May be expressed in this manner (Shorthand)
+```json
+[
+    {
+        "or": false,
+        "filters": [
+            ["author", "sw" "Optimus"],
+            ["author", "ew" "Prime"],
+            ["deleted_at", "eq" null, true]
```
Comment (esbenp) [inline README.md]: Missing `,`

File: src/EloquentBuilderTrait.php [hunk 2]
```diff
@@ -165,15 +192,15 @@ protected function applyFilter(Builder $queryBuilder, array $filter, $or = false
                     $method,
                     $clauseOperator,
                     $value,
-                    $operator === 'in'
+                    $clauseOperator // @depreciated. Here for backwards compatibility
```
Comment (esbenp) [inline src/EloquentBuilderTrait.php]: I think you mean `@deprecated` :p

Comment (pmccarren) [inline src/EloquentBuilderTrait.php]: Hah nice catch!

**Comment by pmccarren** [comment]
Awesome! And you're welcome, glad to help! :D

**Comment by pmccarren** [comment]
Hmm I'm gonna look into why tests didn't catch the issue with the previous commit!

**Comment by pmccarren** [comment]
Hey @esbenp! Would it be possible to get this PR landed in a release? Thanks much!

**Comment by esbenp** [comment]
Thanks for reminding me @pmccarren !! :D

**Pull request merged**